app/views/Settings.js

```diff
@@ -53,11 +57,49 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, []);
 
+  // on load setting page, this method will check whether we have remove location permission from app settings.
+  // It will set status "Location Active" only if permission is granted./*tested properly on android device*/
+  const willParticipate = () => {
+    SetStoreData(PARTICIPATE, 'true').then(() => {
+      // Turn of bluetooth for v1
+      //BroadcastingServices.start();
+    });
+    // Check and see if they actually authorized in the system dialog.
+    // If not, stop services and set the state to !isLogging
+    // Fixes tripleblindmarket/private-kit#129
+    BackgroundGeolocation.checkStatus(({ authorization }) => {
+      if (authorization === BackgroundGeolocation.AUTHORIZED) {
+        LocationServices.start();
+        setBGLocationLogging(true);
+        setIsLogging(true);
+      } else if (authorization === BackgroundGeolocation.NOT_AUTHORIZED) {
+        LocationServices.stop();
+        setBGLocationLogging(false);
+        setIsLogging(false);
+        BackgroundTaskServices.stop();
+      }
+    });
+  };
+
+  // this is called when you click on device permission dialog "DENY" & "ALLOW"{ /*tested properly on android device*/}
+  const checkBackgroundStatus = async () => {
+    BackgroundGeolocation.on('authorization', status => {
+      // when you accept the permission
+      if (status === BackgroundGeolocation.AUTHORIZED) {
+        setIsLogging(true);
+      } else {
+        setIsLogging(false);
+      }
+    });
+  };
+
   const locationToggleButtonPressed = async () => {
     try {
       isLogging ? LocationServices.stop() : LocationServices.start();
       await SetStoreData(PARTICIPATE, !isLogging);
       setIsLogging(!isLogging);
+      setBGLocationLogging(true);
+      checkBackgroundStatus();
```

Comment:
This check is not really a check, so it's not worth doing this each time the button is pressed. Keep it once in the initial `useEffect()`